app/controllers/ticket_desks_controller.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class TicketDesksController < ApplicationController
+  after_action :verify_policy_scoped, only: :index, :show
+  skip_after_action :verify_authorized, only: :index, :show
+
+  def index
+    if user_signed_in?
+      @ticket_desks = policy_scope(TicketDesk)
+      render json: @ticket_desks
+    else 
+      render json: @ticket_desk.errors,
+              status: :unprocessable_entity
+  end
+
+  def create
+    if user_signed_in? && user.admin?
```

Comment:
So again, user will be authenticated here anyway and for checking whenever user is admin it would be better to call:
```ruby
authorize  TicketDesk
```
And handle that logic in pundit policy